atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/postgres/PostgresQueryFactory.java

```diff
@@ -45,15 +45,35 @@ public FullQuery getLatestRowQuery(byte[] row,
                                        long ts,
                                        ColumnSelection columns,
                                        boolean includeValue) {
+        if (columns.allColumnsSelected() || columns.getSelectedColumns().size() > 50) {
```

Comment:
the 50 is just some random crap I wrote, dunno if anyone ever requests that much.